### DISCUSSION TITLE
How to make a case insensitive query?

### DISCUSSION THREAD

**Discussion by mawilms** [supabase/supabase#2586] [Questions] [answered]
Hello everyone,

I'm currently using the Dart package of supabase and I'm wondering how to perform a case insensitive query.

The SQL query I want to use is the following:

```sql
select * from products
where lower(products.name) like lower('%data%');
```

The equivalent I used was so far this one:

```dart
Supabase.instance.client
        .from("products")
        .select("*")
        .like("lower(name)", pattern)
        .execute();
```

This query ends in the following error: "Column products.lower does not exist" which makes sense since we use this annotation for joins. 

I haven't found anything in the docs so far.

I appreciate any answer

**Comment by mawilms** [comment, marked as answer]
Ok I found the solution by myself....

You have to use ilike: https://supabase.io/docs/reference/javascript/ilike

Sorry for the inconvenience!

**Reply by jsjoeio** [reply]
Can you interpolate a value?

```javascript
const cityToMatch = "la"
const { data, error } = await supabase
  .from('cities')
  .select('name, country_id')
  .ilike('name', `%${cityToMatch}%`)
```

**Reply by soedirgo** [reply]
@jsjoeio Yes, but you'd use `*` instead of `%`.

**Reply by jsjoeio** [reply]
Ahhh, okay, thanks for mentioning!
